integration-tests/src/test/gate-token.test.ts

```diff
@@ -107,4 +111,39 @@ describe('Test GATE Token', (): void => {
         },
         900_000,
     );
+
+    // This test must run after the end to end test
+    // We are relying on the gate contract already being set, acceptable assumption since we run tests with --runInBand
```

Comment:
I think this requirement is no longer needed b/c the gate contract is now set in `beforeAll`.

In addition eventhough we use `--runInBand` I'm not quite sure how reliable that is, however that's a side topic.

FTR: I've started prefixing tests with a numeric prefix if the order matters, e.g. `001 - end-to-end` inside collect-coins.

Reply:
I misspoke in the comments. We need the gate contract to be set AND the faucet account to be set. The first sub-test of the e2e test is whether the extrinsic correctly reports that the faucet hasn't yet been set. After the error has been reported we set the faucet and try again this time expecting a success. So we need that sub-test to run before the example